src/pybind/mgr/rgw/module.py

```diff
@@ -55,11 +86,23 @@ def list_daemons(self, service_name, daemon_type=None, daemon_id=None, host=None
         return orchestrator.raise_if_exception(completion)
 
 
+def check_orchestrator(func: FuncT) -> FuncT:
+    @functools.wraps(func)
+    def wrapper(self: Any, *args: Any, **kwargs: Any) -> HandleCommandResult:
+        available = self.api.status()['available']
+        if available:
+            return func(self, *args, **kwargs)
+        else:
+            err_msg = "Cephadm is not available. Please enable cephadm by 'ceph mgr module enable cephadm'."
```

Comment:
the previous version of this getting removed started the error message with `Orchestrator is not available`. Is this explicitly checking for the cephadm mgr module?

Reply:
> 

Well, since right now this only works with cephadm I wanted to make it explicit. Otherwise, a user may think that any orchestrator can handle it which is not correct at this moment. In case in the future we add support for this in rook for example, then we can make the error more generic. WDYT?

Reply:
I guess it's fine for now. Generally I like to keep to saying orchestrator but you have a pointt that cephadm is the only orchestrator that works with this bit currently.